### ISSUE TITLE
trap_FS_FOpenFile fails to open file not in pk3 after mapchange, when sv_pure is enabled

### ISSUE THREAD

**Issue by mecwerks** [zturtleman/spearmint#282]
Pretty specific, but I came across this issue when trying to write a custom mapcycle config.
On the first map load, "mapcycle.txt" is loaded fine. But after the map changes the file fails to load. But only if sv_pure is enabled. 

Function I'm currently using to load the file (copy/paste of G_LoadBotsFromFile() mostly)
```
char *g_mapList[1024];
int g_numMaps;

void G_LoadMapList(void)
{
	int				len;
	fileHandle_t	f;
	char			buf[MAX_ARENAS_TEXT];
	char *filename = "mapcycle.txt";

	len = trap_FS_FOpenFile(filename, &f, FS_READ);

	if ( !f ) {
		trap_Print(va(S_COLOR_RED "file not found: %s\n", filename));
		return;
	}
	if ( len >= MAX_ARENAS_TEXT ) {
		trap_FS_FCloseFile( f );
		trap_Print(va(S_COLOR_RED "file too large: %s is %i, max allowed is %i\n", filename, len, MAX_ARENAS_TEXT));
		return;
	}

	trap_FS_Read( buf, len, f );
	buf[len] = 0;
	trap_FS_FCloseFile( f );

	g_numMaps += G_ParseInfos(buf, MAX_ARENAS - g_numMaps, &g_mapList[g_numMaps]);

	G_Printf("Found %d maps for mapcycle.\n", g_numMaps);
}
```

Example of mapcycle.txt:
```
{
  "map"  "q3dm17"
}
{
  "map" "q3tourney1"
  "capturelimit" "24"
}
```

Putting the file in a pk3 file or setting sv_pure to 0 makes it load correctly every time.

The best way to test this without making code modifications to load a seperate file would be moving scripts/arenas.txt or scripts/bots.txt from pak0.pk3 to baseq3/scripts/, removing other pak files except one for spearmint vms, and then setting mint-game.settings accordingly. It should reproduce the same issue I'm having.

What would you want to happen in a situation like this? 

Seems like it could be a bug to let cgame load files from outside of pakfiles when sv_pure is enabled. could allow people to have modified teamarena ui files, and model skin/animation files. (not sure if that could really cause an issue though.)

**Comment by zturtleman** [comment]
Okay I've added ".txt" to the [unzipped file extensions](https://github.com/zturtleman/spearmint/blob/c3a8c4ec82aba71e90aa7e2e30b257957947ba7e/code/qcommon/files.c#L1444-L1450) that can be accessed in pure mode. Unzipped Team Arena menus are allowed which can set sv_cheats to 0 in quake3/ioquake3 to allow using all cheat cvars :confetti_ball:.

Adding ".txt" makes a lot of sense for supporting custom Team Arena huds (cg_hudFiles "ui/hud.txt" cvar), bots.txt (g_botsFile "scripts/bots.txt" cvar), arenas.txt (g_arenasFile "scripts/arenas.txt cvar), etc. In the past I've used .cfg as a workaround.

---

The initial Game VM load always acts as if pure mode is disabled. Pure mode ends up affecting Game VM after client connects and the client enables pure mode. Dedicated server never uses pure mode. This issue exists in vanilla quake3 and ioquake3 too.

Pure mode also reorders the search paths so that pk3s are searched before unzipped files. This can cause a unzipped file to be used but then after map change use the file from a pk3. (Only affects ioquake3 and Spearmint. Vanilla Quake 3 always searches pk3s before unzipped.)

For consistency the solution is probably enabling pure mode restrictions/order (see fs_numServerPaks in code/qcommon/files.c) in SV_SpawnServer() before FS_Restart() and the initial Game VM load.
